### DISCUSSION TITLE
Simplification of very complex flexible model

### DISCUSSION THREAD

**Discussion by NemoVis** [jgerstmayr/EXUDYN#81] [General]
Hello!
   I'm in the process of numerical experiments for my PhD that comprises some flexible structures. The structures I need are some complex (rather big) frames with long flexible elements and rigid but heavy connectors. As the flexible elements may be quite different and the beam approximation is not always appropriate so I use FEA output (for now just NetGen) to generate the flexible body (off-cause with the Craig-Bampton reduction to make the model computationally feasible). Though each of the flexible elements after reduction has very few degrees of freedom the whole structure seems to be enormously big and it seems that the constraints that I use extensively are the reason. If one use the classical modal synthesis by Craig-Bampton the aforementioned degrees of freedom should collapse and no constraint equations would appear. 
     So my question is weather it's possible to transform the model with multiple heavily interconnected flexible and rigid bodies into one single flexible body in EXUDYN and how? May be there are some guide lines on how to modify the source code (like creating my own element) to solve this?
     
     Thanks in advance

**Comment by jgerstmayr** [comment]
If I understood correctly, you only like to model linear elastic bodies as one single linear body. In this case, you can set up the constraints (either at nodal level with SphericalJoints or with MarkerSuperElementRigid and GenericJoint). Then you export the systems linear equations of motion with exudyns functionality and project into the constraint nullspace (see computation of eigenmodes).

**Reply by NemoVis** [reply]
Thanks for your quick response!
Yep, I've also thought about leaving only linear part but I didn't know that EXUDYN can perform null-space projection by default. 
Unfortunately some of the spacial structure, I'm studying, undergoes general (rigid body motion) along with deformations (like rolling on the ground or free fall, or just dangling in zero gravity). These are not so mandatory but would be preferable in case the questions like "Does your  algorithm works in general motion case?" have to be tackled.
